File: banco_problemas_aplicados/nivel_intermedio/soluciones/c/problema32.c
/*
 * Problema 32 - Conteo de Nodos con Voltaje Critico
 * Contexto: matriz de tensiones. Cuenta cuantas celdas superan un umbral de sobretension.
 */
#include <stdio.h>

int cuentaCriticos(int n, int m, double mat[n][m], double umbral) {
    int contador = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            if (mat[i][j] > umbral) {
                contador = contador + 1;
            }
        }
    }
    return contador;
}

int main(void) {
    int n, m;
    double umbral;
    printf("Numero de filas (N): ");
    scanf("%d", &n);
    printf("Numero de columnas (M): ");
    scanf("%d", &m);

    double mat[n][m];
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            printf("mat[%d][%d]: ", i, j);
            scanf("%lf", &mat[i][j]);
        }
    }

    printf("Umbral critico de sobretension: ");
    scanf("%lf", &umbral);

    printf("Nodos que superan el umbral: %d\n", cuentaCriticos(n, m, mat, umbral));
    return 0;
}

File: banco_problemas_aplicados/nivel_intermedio/soluciones/java/Problema32.java
/*
 * Problema 32 - Conteo de Nodos con Voltaje Critico
 * Cuenta cuantas celdas de la matriz superan un umbral.
 */
import java.util.Scanner;

public class Problema32 {

    static int cuentaCriticos(double[][] mat, double umbral) {
        int contador = 0;
        for (int i = 0; i < mat.length; i++) {
            for (int j = 0; j < mat[i].length; j++) {
                if (mat[i][j] > umbral) {
                    contador = contador + 1;
                }
            }
        }
        return contador;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Numero de filas (N): ");
        int n = sc.nextInt();
        System.out.print("Numero de columnas (M): ");
        int m = sc.nextInt();

        double[][] mat = new double[n][m];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                System.out.print("mat[" + i + "][" + j + "]: ");
                mat[i][j] = sc.nextDouble();
            }
        }

        System.out.print("Umbral critico de sobretension: ");
        double umbral = sc.nextDouble();

        System.out.println("Nodos que superan el umbral: " + cuentaCriticos(mat, umbral));
        sc.close();
    }
}
